src/run.js

```diff
@@ -0,0 +1,116 @@
+const cozyClient = require('cozy-konnector-libs/dist/libs/cozyclient')
+const errors = require('cozy-konnector-libs/dist/helpers/errors')
+const { log } = require('cozy-konnector-libs')
+
+const {
+  fetchAndSaveTrips,
+  fetchAndSaveManualEntries,
+  fetchTripsMetadata
+} = require('./lib')
+const { getFirstAndLastTripTimestamp } = require('./trace-requests')
+const {
+  canSaveNextTripsChunk,
+  restartKonnector,
+  createChunks
+} = require('./utils')
+const { saveAccountData } = require('./save')
+const { TRIPS_CHUNK_SIZE } = require('./const')
+const client = cozyClient.new
+
+const getTimeout = () => {
+  const maxExecutionTimeSeconds = parseInt(process.env.COZY_TIME_LIMIT, 10)
+  return maxExecutionTimeSeconds - 100
+}
+
+const startExecTime = new Date()
+
+const run = async ({ fields, accountData, accountId }) => {
+  log('info', 'Start the Tracemob konnector')
+
+  // We use the login field as an identifier that can be used as a device name.
+  const device = fields.login || 'Tracemob'
+  // TODO: the token might be retrieved directly in the account
+  const token = fields.password
+
+  /* Get the trips starting date */
+  let startDate
+  let startManualDate
+  let firstRun = false
+  try {
+    if (accountData && accountData.lastSavedTripDate) {
+      startDate = new Date(accountData.lastSavedTripDate)
+    }
+    if (accountData && accountData.lastSavedManualDate) {
+      startManualDate = new Date(accountData.lastSavedManualDate)
+    }
+    if (!startDate) {
+      const timestamps = await getFirstAndLastTripTimestamp(token)
+      if (!timestamps.start_ts || !timestamps.end_ts) {
+        log('info', 'No trip saved yet. Abort.')
+        return
+      }
+      startDate = new Date(timestamps.start_ts * 1000)
+      firstRun = true
+    }
+    if (!startManualDate) {
+      startManualDate = startDate
+    }
+
+    /* Extract the days having saved trips */
+    log('info', `Fetch trips metadata from ${startDate.toISOString()}`)
+    const tripsMetadata = await fetchTripsMetadata(token, startDate, {
+      excludeFirst: !firstRun
```

Comment:
Not sure to understand this (based on `getServerCollectionFromDate` code), is that because the datastream date limits are both inclusives?

Reply:
Yes, when we query the trace server, the dates are inclusive. As we start from last retrieved date, we want to exclude the first result (except when it's the first run).